src/qt/walletmodel.cpp

```diff
@@ -514,6 +514,8 @@ bool WalletModel::bumpFee(uint256 hash, uint256& new_hash)
     questionString.append("</td><td>");
     questionString.append(BitcoinUnits::formatHtmlWithUnit(getOptionsModel()->getDisplayUnit(), new_fee));
     questionString.append("</td></tr></table>");
+    questionString.append("<br /><br />");
+    questionString.append(tr("WARNING: New inputs and outputs can be added in replacement transaction if necessary which affects privacy"));
```

Comment:
nit: Could a full stop at the sentence end make it better?

Reply:
This seems to confusing to normal people. They don't know what inputs or outputs are, nor necessarily care how Bitcoin works internally.

Reply:
> This seems to confusing to normal people.

:D

> They don't know what inputs or outputs are, nor necessarily care how Bitcoin works internally.

Educational purpose, no?

Could it be re-worded without revealing of internal tx details?

Reply:
IMO best way to rework this is, if expert options for coin control features is enabled then

-  Show proper internal TX details or a warning message in dialog like this.

@prayank23  @luke-jr @hebasto

Reply:
> nit: Could a full stop at the sentence end make it better?

Done

> This seems to confusing to normal people. They don't know what inputs or outputs are, nor necessarily care how Bitcoin works internally.

"Inputs" is already mentioned in GUI in 'Coin Control Features", user who did the initial transaction by selecting inputs obviously cares about it and "Outputs" can be replaced with "Change Address" although it can mean different things but easier to understand.

Reply:
Users also shouldn't *need* to know about internals.

Coin Control Features are an advanced feature.

Maybe just only append this when it's enabled?

Reply:
> Maybe just only append this when it's enabled?

This makes sense. What is the best way to know if coin control was used for a transaction? So that it can be used in `if` statement for this warning.

Reply:
AFAIK it isn't possible right now, we'd just need to assume the current checkbox state is what the user wants behaviour based on

Reply:
IMO a generic warning message based on `bumpfee` docs is good enough for now. In long term we need to fix few other issues related to coin selection algorithm, coin control for RBF etc. It can add random inputs even when coin control is disabled.

Reply:
With CC disabled, it was *created* with random inputs already...

Reply:
Added `if` statement to check if CC is enabled